// moxfield-api/dist/moxfield-api.cjs
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/moxfield-api.ts
var moxfield_api_exports = {};
__export(moxfield_api_exports, {
  AffiliatesSchema: () => affiliates_schema_default,
  AuthorSchema: () => author_schema_default,
  AuthorTagsSchema: () => author_tags_schema_default,
  BoardCardSchema: () => board_card_schema_default,
  BoardSchema: () => board_schema_default,
  BoardTypeSchema: () => board_type_schema_default,
  BoardsSchema: () => boards_schema_default,
  CardFaceSchema: () => card_face_schema_default,
  CardSchema: () => card_schema_default,
  CardsToTokensSchema: () => cards_to_tokens_schema_default,
  ColorIdentityPercentagesSchema: () => color_identity_percentages_schema_default,
  ColorIdentitySchema: () => color_identity_schema_default,
  ColorPercentagesSchema: () => color_percentages_schema_default,
  DeckListSchema: () => deck_list_schema_default,
  FinishSchema: () => finish_schema_default,
  FolderSchema: () => folder_schema_default,
  HubSchema: () => hub_schema_default,
  LegalitiesSchema: () => legalities_schema_default,
  MeldPartSchema: () => meld_part_schema_default,
  MeldResultSchema: () => meld_result_schema_default,
  MoxfieldError: () => MoxfieldError,
  NotFoundMoxfieldError: () => NotFoundMoxfieldError,
  OriginalDeckSchema: () => original_deck_schema_default,
  PricesSchema: () => prices_schema_default,
  PrintCardSchema: () => print_card_schema_default,
  RaritySchema: () => rarity_schema_default,
  SetTypeSchema: () => set_type_schema_default,
  TokenMappingsSchema: () => token_mappings_schema_default,
  TokensToCardsSchema: () => tokens_to_cards_schema_default,
  ViewSettingsSchema: () => view_settings_schema_default,
  VisibilitySchema: () => visibility_schema_default,
  default: () => api_default
});
module.exports = __toCommonJS(moxfield_api_exports);

// src/api/deck-list/deck-list.api.ts
var import_zod31 = require("zod");
var import_zod_fetch = require("zod-fetch");

// src/error/moxfield.error.ts
var MoxfieldError = class extends Error {
  constructor(status, message) {
    super(message);
    this.status = status;
  }
};

// src/error/not-found.error.ts
var NotFoundMoxfieldError = class extends MoxfieldError {
  constructor(message) {
    super(404, message);
  }
};

// src/fetcher/moxfield.fetcher.ts
function createMoxfieldFetcher(fetcher = fetch) {
  return async function moxfieldFetcher(...arguments_) {
    const response = await fetcher(...arguments_);
    if (response.ok) {
      return response.json();
    }
    if (response.status === 404) {
      throw new NotFoundMoxfieldError();
    }
    const object = await response.json();
    const status = "status" in object ? Number(object.status) : void 0;
    const error = "error" in object ? String(object.error) : void 0;
    throw new MoxfieldError(status, error);
  };
}

// src/schema/affiliates/affiliates.schema.ts
var import_zod2 = require("zod");

// src/helpers/is-test.ts
function isTest() {
  return typeof process !== "undefined" && process.env.NODE_ENV === "test";
}

// src/helpers/strict.ts
var import_zod = __toESM(require("zod"), 1);
function conditionalStrict(schema) {
  if (isTest()) {
    return import_zod.default.strictObject(schema.shape);
  }
  return schema;
}

// src/schema/affiliates/affiliates.schema.ts
var AffiliatesSchema = import_zod2.z.object({
  ch: import_zod2.z.string(),
  ck: import_zod2.z.string(),
  cm: import_zod2.z.string(),
  csi: import_zod2.z.string(),
  ct: import_zod2.z.string(),
  scg: import_zod2.z.string(),
  tcg: import_zod2.z.string()
});
var affiliates_schema_default = conditionalStrict(AffiliatesSchema);

// src/schema/author/author.schema.ts
var import_zod3 = require("zod");
var AuthorSchema = import_zod3.z.object({
  badges: import_zod3.z.array(import_zod3.z.string()),
  displayName: import_zod3.z.string(),
  profileImageUrl: import_zod3.z.url().nullish(),
  promos: import_zod3.z.array(import_zod3.z.unknown()).nullish(),
  userName: import_zod3.z.string()
});
var author_schema_default = conditionalStrict(AuthorSchema);

// src/schema/author-tags/author-tags.schema.ts
var import_zod4 = require("zod");
var AuthorTagsSchema = import_zod4.z.record(import_zod4.z.string(), import_zod4.z.array(import_zod4.z.string()));
var author_tags_schema_default = AuthorTagsSchema;

// src/schema/board/board.schema.ts
var import_zod18 = require("zod");

// src/schema/board-card/board-card.schema.ts
var import_zod17 = require("zod");

// src/schema/board-type/board-type.schema.ts
var import_zod5 = require("zod");
var BoardTypesSchema = import_zod5.z.union([
  import_zod5.z.literal("attractions"),
  import_zod5.z.literal("commanders"),
  import_zod5.z.literal("companions"),
  import_zod5.z.literal("contraptions"),
  import_zod5.z.literal("mainboard"),
  import_zod5.z.literal("maybeboard"),
  import_zod5.z.literal("partners"),
  import_zod5.z.literal("planes"),
  import_zod5.z.literal("schemes"),
  import_zod5.z.literal("sideboard"),
  import_zod5.z.literal("signatureSpells"),
  import_zod5.z.literal("stickers"),
  import_zod5.z.literal("tokens")
]);
var board_type_schema_default = BoardTypesSchema;

// src/schema/card/card.schema.ts
var import_zod14 = require("zod");

// src/schema/card-face/card-face.schema.ts
var import_zod7 = require("zod");

// src/schema/color-identity/color-identity.schema.ts
var import_zod6 = require("zod");
var ColorIdentitySchema = import_zod6.z.union([import_zod6.z.literal("W"), import_zod6.z.literal("U"), import_zod6.z.literal("B"), import_zod6.z.literal("R"), import_zod6.z.literal("G")]);
var color_identity_schema_default = ColorIdentitySchema;

// src/schema/card-face/card-face.schema.ts
var CardFaceSchema = import_zod7.z.object({
  color_indicator: import_zod7.z.array(color_identity_schema_default),
  colors: import_zod7.z.array(color_identity_schema_default),
  defense: import_zod7.z.string().nullish(),
  flavor_name: import_zod7.z.string().nullish(),
  flavor_text: import_zod7.z.string().nullish(),
  id: import_zod7.z.string(),
  image_seq: import_zod7.z.number(),
  loyalty: import_zod7.z.string().nullish(),
  mana_cost: import_zod7.z.string().nullish(),
  name: import_zod7.z.string(),
  oracle_text: import_zod7.z.string().nullish(),
  power: import_zod7.z.string().nullish(),
  toughness: import_zod7.z.string().nullish(),
  type_line: import_zod7.z.string()
});
var card_face_schema_default = conditionalStrict(CardFaceSchema);

// src/schema/legalities/legalities.schema.ts
var import_zod8 = require("zod");
var LegalitiesFormatSchema = import_zod8.z.union([
  import_zod8.z.literal("alchemy"),
  import_zod8.z.literal("brawl"),
  import_zod8.z.literal("commander"),
  import_zod8.z.literal("duel"),
  import_zod8.z.literal("explorer"),
  import_zod8.z.literal("future"),
  import_zod8.z.literal("gladiator"),
  import_zod8.z.literal("historic"),
  import_zod8.z.literal("historicbrawl"),
  import_zod8.z.literal("legacy"),
  import_zod8.z.literal("modern"),
  import_zod8.z.literal("oathbreaker"),
  import_zod8.z.literal("oldschool"),
  import_zod8.z.literal("pauper"),
  import_zod8.z.literal("paupercommander"),
  import_zod8.z.literal("penny"),
  import_zod8.z.literal("pioneer"),
  import_zod8.z.literal("predh"),
  import_zod8.z.literal("premodern"),
  import_zod8.z.literal("standard"),
  import_zod8.z.literal("standardbrawl"),
  import_zod8.z.literal("timeless"),
  import_zod8.z.literal("vintage")
]);
var LegalitiesValueSchema = import_zod8.z.union([
  import_zod8.z.literal("legal"),
  import_zod8.z.literal("not_legal"),
  import_zod8.z.literal("banned"),
  import_zod8.z.literal("restricted")
]);
var LegalitiesSchema = import_zod8.z.partialRecord(LegalitiesFormatSchema, LegalitiesValueSchema);
var legalities_schema_default = LegalitiesSchema;

// src/schema/meld-part/meld-part.schema.ts
var import_zod9 = require("zod");
var MeldPartSchema = import_zod9.z.object({
  id: import_zod9.z.string(),
  imageCardId: import_zod9.z.string(),
  imageCardIdIsCardFace: import_zod9.z.boolean(),
  name: import_zod9.z.string(),
  uniqueCardId: import_zod9.z.string()
});
var meld_part_schema_default = conditionalStrict(MeldPartSchema);

// src/schema/meld-result/meld-result.schema.ts
var import_zod10 = require("zod");
var MeldResultSchema = import_zod10.z.object({
  id: import_zod10.z.string(),
  imageCardId: import_zod10.z.string(),
  imageCardIdIsCardFace: import_zod10.z.boolean(),
  name: import_zod10.z.string(),
  uniqueCardId: import_zod10.z.string()
});
var meld_result_schema_default = conditionalStrict(MeldResultSchema);

// src/schema/prices/prices.schema.ts
var import_zod11 = require("zod");
var PricesSchema = import_zod11.z.object({
  ck: import_zod11.z.number().nullish(),
  ck_buy: import_zod11.z.number().nullish(),
  ck_buy_etched: import_zod11.z.number().nullish(),
  ck_buy_etched_qty: import_zod11.z.number().nullish(),
  ck_buy_foil: import_zod11.z.number().nullish(),
  ck_buy_foil_qty: import_zod11.z.number().nullish(),
  ck_buy_qty: import_zod11.z.number().nullish(),
  ck_etched: import_zod11.z.number().nullish(),
  ck_foil: import_zod11.z.number().nullish(),
  csi: import_zod11.z.number().nullish(),
  csi_buy: import_zod11.z.number().nullish(),
  csi_buy_foil: import_zod11.z.number().nullish(),
  csi_buy_foil_qty: import_zod11.z.number().nullish(),
  csi_buy_qty: import_zod11.z.number().nullish(),
  csi_foil: import_zod11.z.number().nullish(),
  ct: import_zod11.z.number().nullish(),
  ct_etched: import_zod11.z.number().nullish(),
  ct_foil: import_zod11.z.number().nullish(),
  eur: import_zod11.z.number().nullish(),
  eur_foil: import_zod11.z.number().nullish(),
  lastUpdatedAtUtc: import_zod11.z.coerce.date(),
  mp: import_zod11.z.number().nullish(),
  mp_etched: import_zod11.z.number().nullish(),
  mp_foil: import_zod11.z.number().nullish(),
  mp_qty: import_zod11.z.number().nullish(),
  scg: import_zod11.z.number().nullish(),
  scg_buy: import_zod11.z.number().nullish(),
  scg_etched: import_zod11.z.number().nullish(),
  scg_etched_buy: import_zod11.z.number().nullish(),
  scg_etched_qty: import_zod11.z.number().nullish(),
  scg_foil: import_zod11.z.number().nullish(),
  scg_foil_buy: import_zod11.z.number().nullish(),
  scg_foil_qty: import_zod11.z.number().nullish(),
  scg_qty: import_zod11.z.number().nullish(),
  tix: import_zod11.z.number().nullish(),
  usd: import_zod11.z.number().nullish(),
  usd_etched: import_zod11.z.number().nullish(),
  usd_foil: import_zod11.z.number().nullish()
});
var prices_schema_default = conditionalStrict(PricesSchema);

// src/schema/rarity/rarity.schema.ts
var import_zod12 = require("zod");
var RaritySchema = import_zod12.z.union([
  import_zod12.z.literal("bonus"),
  import_zod12.z.literal("common"),
  import_zod12.z.literal("mythic"),
  import_zod12.z.literal("rare"),
  import_zod12.z.literal("special"),
  import_zod12.z.literal("uncommon")
]);
var rarity_schema_default = RaritySchema;

// src/schema/set-type/set-type.schema.ts
var import_zod13 = require("zod");
var SetTypeSchema = import_zod13.z.union([
  import_zod13.z.literal("alchemy"),
  import_zod13.z.literal("archenemy"),
  import_zod13.z.literal("arsenal"),
  import_zod13.z.literal("box"),
  import_zod13.z.literal("commander"),
  import_zod13.z.literal("core"),
  import_zod13.z.literal("draft_innovation"),
  import_zod13.z.literal("duel_deck"),
  import_zod13.z.literal("eternal"),
  import_zod13.z.literal("expansion"),
  import_zod13.z.literal("from_the_vault"),
  import_zod13.z.literal("funny"),
  import_zod13.z.literal("masterpiece"),
  import_zod13.z.literal("masters"),
  import_zod13.z.literal("memorabilia"),
  import_zod13.z.literal("minigame"),
  import_zod13.z.literal("planechase"),
  import_zod13.z.literal("premium_deck"),
  import_zod13.z.literal("promo"),
  import_zod13.z.literal("spellbook"),
  import_zod13.z.literal("starter"),
  import_zod13.z.literal("token"),
  import_zod13.z.literal("treasure_chest"),
  import_zod13.z.literal("vanguard")
]);
var set_type_schema_default = SetTypeSchema;

// src/schema/card/card.schema.ts
var CardSchema = import_zod14.z.object({
  acorn: import_zod14.z.boolean(),
  allRaritiesMask: import_zod14.z.number().nullish(),
  arena_id: import_zod14.z.number().nullish(),
  artist: import_zod14.z.string().nullish(),
  attraction_lights: import_zod14.z.array(import_zod14.z.number()).nullish(),
  border_color: import_zod14.z.string(),
  card_faces: import_zod14.z.array(card_face_schema_default),
  cardHoarderUrl: import_zod14.z.url().nullish(),
  cardkingdom_etched_id: import_zod14.z.number().nullish(),
  cardkingdom_foil_id: import_zod14.z.number().nullish(),
  cardkingdom_id: import_zod14.z.number().nullish(),
  cardKingdomEtchedUrl: import_zod14.z.string().nullish(),
  cardKingdomFoilUrl: import_zod14.z.string().nullish(),
  cardKingdomUrl: import_zod14.z.string().nullish(),
  cardmarket_id: import_zod14.z.number().nullish(),
  cardMarketUrl: import_zod14.z.url().nullish(),
  cardTraderEtchedUrl: import_zod14.z.url().nullish(),
  cardTraderFoilUrl: import_zod14.z.url().nullish(),
  cardTraderUrl: import_zod14.z.url().nullish(),
  cmc: import_zod14.z.number(),
  cn: import_zod14.z.string(),
  color_identity: import_zod14.z.array(color_identity_schema_default),
  color_indicator: import_zod14.z.array(color_identity_schema_default),
  colors: import_zod14.z.array(color_identity_schema_default),
  colorshifted: import_zod14.z.boolean(),
  content_warning: import_zod14.z.boolean(),
  coolStuffIncFoilUrl: import_zod14.z.url().nullish(),
  coolStuffIncUrl: import_zod14.z.url().nullish(),
  defaultFinish: import_zod14.z.string(),
  digital: import_zod14.z.boolean(),
  edhrec_rank: import_zod14.z.number().nullish(),
  etched: import_zod14.z.boolean(),
  flavor_name: import_zod14.z.string().nullish(),
  flavor_text: import_zod14.z.string().nullish(),
  foil: import_zod14.z.boolean(),
  frame: import_zod14.z.string().nullish(),
  frame_effects: import_zod14.z.array(import_zod14.z.string()).nullish(),
  glossy: import_zod14.z.boolean(),
  has_arena_legal: import_zod14.z.boolean(),
  has_multiple_editions: import_zod14.z.boolean(),
  id: import_zod14.z.string(),
  image_seq: import_zod14.z.number(),
  isArenaLegal: import_zod14.z.boolean(),
  isCovered: import_zod14.z.boolean(),
  isPauperCommander: import_zod14.z.boolean(),
  isToken: import_zod14.z.boolean(),
  lang: import_zod14.z.string(),
  latest: import_zod14.z.boolean(),
  layout: import_zod14.z.string(),
  legalities: legalities_schema_default,
  loyalty: import_zod14.z.string().nullish(),
  mana_cost: import_zod14.z.string().nullish(),
  manapool_url: import_zod14.z.url().nullish(),
  meld_part: meld_part_schema_default.nullish(),
  meld_parts: import_zod14.z.array(meld_part_schema_default).nullish(),
  meld_result: meld_result_schema_default.nullish(),
  mtgo_id: import_zod14.z.number().nullish(),
  multiverse_ids: import_zod14.z.array(import_zod14.z.number()).nullish(),
  name: import_zod14.z.string(),
  nonfoil: import_zod14.z.boolean(),
  oracle_text: import_zod14.z.string().nullish(),
  power: import_zod14.z.string().nullish(),
  prices: prices_schema_default,
  printed_name: import_zod14.z.string().nullish(),
  printed_text: import_zod14.z.string().nullish(),
  printed_type_line: import_zod14.z.string().nullish(),
  promo_types: import_zod14.z.array(import_zod14.z.string()),
  rarity: rarity_schema_default,
  released_at: import_zod14.z.coerce.date().nullish(),
  reprint: import_zod14.z.boolean(),
  reserved: import_zod14.z.boolean(),
  scryfall_id: import_zod14.z.uuid(),
  set: import_zod14.z.string(),
  set_name: import_zod14.z.string(),
  set_type: set_type_schema_default.nullish(),
  starcitygames_etched_sku: import_zod14.z.string().nullish(),
  starcitygames_etched_url: import_zod14.z.url().nullish(),
  starcitygames_foil_sku: import_zod14.z.string().nullish(),
  starcitygames_foil_url: import_zod14.z.url().nullish(),
  starcitygames_sku: import_zod14.z.string().nullish(),
  starcitygames_url: import_zod14.z.url().nullish(),
  tcgplayer_id: import_zod14.z.number().nullish(),
  tcgPlayerUrl: import_zod14.z.url().nullish(),
  toughness: import_zod14.z.string().nullish(),
  type: import_zod14.z.string().nullish(),
  type_line: import_zod14.z.string().nullish(),
  uniqueCardId: import_zod14.z.string()
});
var card_schema_default = conditionalStrict(CardSchema);

// src/schema/finish/finish.schema.ts
var import_zod15 = require("zod");
var FinishSchema = import_zod15.z.union([import_zod15.z.literal("nonFoil"), import_zod15.z.literal("foil"), import_zod15.z.literal("etched"), import_zod15.z.literal("glossy")]);
var finish_schema_default = FinishSchema;

// src/schema/print-card/print-card.schema.ts
var import_zod16 = require("zod");
var PrintCardSchema = import_zod16.z.object({
  card: card_schema_default,
  cardHoarderUrl: import_zod16.z.url().nullish(),
  cardKingdomFoilUrl: import_zod16.z.string().nullish(),
  cardKingdomUrl: import_zod16.z.string().nullish(),
  cardMarketUrl: import_zod16.z.url().nullish(),
  cn: import_zod16.z.string(),
  coolStuffIncFoilUrl: import_zod16.z.url().nullish(),
  coolStuffIncUrl: import_zod16.z.url().nullish(),
  finish: finish_schema_default,
  foil: import_zod16.z.boolean(),
  hash: import_zod16.z.string(),
  id: import_zod16.z.string(),
  imageCardId: import_zod16.z.string(),
  imageCardIdIsCardFace: import_zod16.z.boolean(),
  isAlter: import_zod16.z.boolean(),
  isFoil: import_zod16.z.boolean(),
  isProxy: import_zod16.z.boolean(),
  lang: import_zod16.z.string(),
  nonfoil: import_zod16.z.boolean(),
  prices: prices_schema_default,
  quantity: import_zod16.z.number(),
  rarity: rarity_schema_default,
  set: import_zod16.z.string(),
  set_name: import_zod16.z.string(),
  tcgPlayerUrl: import_zod16.z.url().nullish()
});
var print_card_schema_default = conditionalStrict(PrintCardSchema);

// src/schema/board-card/board-card.schema.ts
var BoardCardSchema = import_zod17.z.object({
  boardType: board_type_schema_default,
  card: card_schema_default,
  cmcOverride: import_zod17.z.number().nullish(),
  colorIdentityOverride: import_zod17.z.array(color_identity_schema_default).nullish(),
  excludedFromColor: import_zod17.z.boolean(),
  finish: finish_schema_default,
  isAlter: import_zod17.z.boolean(),
  isCompanion: import_zod17.z.boolean().nullish(),
  isFoil: import_zod17.z.boolean(),
  isProxy: import_zod17.z.boolean(),
  manaCostOverride: import_zod17.z.string().nullish(),
  printingData: import_zod17.z.array(print_card_schema_default).nullish(),
  quantity: import_zod17.z.number(),
  useCmcOverride: import_zod17.z.boolean(),
  useColorIdentityOverride: import_zod17.z.boolean(),
  useManaCostOverride: import_zod17.z.boolean()
});
var board_card_schema_default = conditionalStrict(BoardCardSchema);

// src/schema/board/board.schema.ts
var BoardSchema = import_zod18.z.object({
  cards: import_zod18.z.record(import_zod18.z.string(), board_card_schema_default),
  count: import_zod18.z.number()
});
var board_schema_default = conditionalStrict(BoardSchema);

// src/schema/boards/boards.schema.ts
var import_zod19 = require("zod");
var BoardTypesSchema2 = import_zod19.z.union([
  import_zod19.z.literal("attractions"),
  import_zod19.z.literal("commanders"),
  import_zod19.z.literal("companions"),
  import_zod19.z.literal("contraptions"),
  import_zod19.z.literal("mainboard"),
  import_zod19.z.literal("maybeboard"),
  import_zod19.z.literal("planes"),
  import_zod19.z.literal("schemes"),
  import_zod19.z.literal("sideboard"),
  import_zod19.z.literal("signatureSpells"),
  import_zod19.z.literal("stickers"),
  import_zod19.z.literal("tokens")
]);
var BoardsSchema = import_zod19.z.record(BoardTypesSchema2, board_schema_default);
var boards_schema_default = BoardsSchema;

// src/schema/cards-to-tokens/cards-to-tokens.schema.ts
var import_zod20 = require("zod");
var CardsToTokensSchema = import_zod20.z.record(import_zod20.z.string(), import_zod20.z.array(import_zod20.z.string()));
var cards_to_tokens_schema_default = CardsToTokensSchema;

// src/schema/color-identity-percentages/color-identity-percentages.schema.ts
var import_zod21 = require("zod");
var ColorIdentityPercentagesSchema = import_zod21.z.object({
  black: import_zod21.z.number(),
  blue: import_zod21.z.number(),
  green: import_zod21.z.number(),
  red: import_zod21.z.number(),
  white: import_zod21.z.number()
});
var color_identity_percentages_schema_default = conditionalStrict(ColorIdentityPercentagesSchema);

// src/schema/color-percentages/color-percentages.schema.ts
var import_zod22 = require("zod");
var ColorPercentagesSchema = import_zod22.z.object({
  black: import_zod22.z.number(),
  blue: import_zod22.z.number(),
  green: import_zod22.z.number(),
  red: import_zod22.z.number(),
  white: import_zod22.z.number()
});
var color_percentages_schema_default = conditionalStrict(ColorPercentagesSchema);

// src/schema/deck-list/deck-list.schema.ts
var import_zod30 = require("zod");

// src/schema/folder/folder.schema.ts
var import_zod23 = require("zod");
var FolderSchema = import_zod23.z.object({
  id: import_zod23.z.string(),
  name: import_zod23.z.string()
});
var folder_schema_default = conditionalStrict(FolderSchema);

// src/schema/hub/hub.schema.ts
var import_zod24 = require("zod");
var HubSchema = import_zod24.z.object({
  description: import_zod24.z.string(),
  name: import_zod24.z.string()
});
var hub_schema_default = conditionalStrict(HubSchema);

// src/schema/original-deck/original-deck.schema.ts
var import_zod26 = require("zod");

// src/schema/visibility/visibility.schema.ts
var import_zod25 = require("zod");
var VisibilitySchema = import_zod25.z.union([
  import_zod25.z.literal("unlisted"),
  import_zod25.z.literal("private"),
  import_zod25.z.literal("public"),
  import_zod25.z.literal("deleted")
]);
var visibility_schema_default = VisibilitySchema;

// src/schema/original-deck/original-deck.schema.ts
var OriginalDeckSchema = import_zod26.z.object({
  authors: import_zod26.z.array(author_schema_default),
  createdAtUtc: import_zod26.z.coerce.date(),
  createdByUser: author_schema_default,
  id: import_zod26.z.string(),
  lastUpdatedAtUtc: import_zod26.z.coerce.date(),
  mainCardId: import_zod26.z.string().nullish(),
  mainCardIdIsBackFace: import_zod26.z.boolean(),
  mainCardIdIsCardFace: import_zod26.z.boolean(),
  name: import_zod26.z.string(),
  publicId: import_zod26.z.string(),
  visibility: visibility_schema_default
});
var original_deck_schema_default = conditionalStrict(OriginalDeckSchema);

// src/schema/token-mappings/token-mappings.schema.ts
var import_zod27 = require("zod");
var TokenMappingsSchema = import_zod27.z.record(import_zod27.z.string(), card_schema_default);
var token_mappings_schema_default = TokenMappingsSchema;

// src/schema/tokens-to-cards/tokens-to-cards.schema.ts
var import_zod28 = require("zod");
var TokensToCardsSchema = import_zod28.z.record(import_zod28.z.string(), import_zod28.z.array(import_zod28.z.string()));
var tokens_to_cards_schema_default = TokensToCardsSchema;

// src/schema/view-settings/view-settings.schema.ts
var import_zod29 = require("zod");
var ViewSettingsSchema = import_zod29.z.object({
  allowMultiplePrintings: import_zod29.z.boolean(),
  groupBy: import_zod29.z.string(),
  sortBy: import_zod29.z.string(),
  useMana: import_zod29.z.boolean(),
  usePrice: import_zod29.z.boolean(),
  useSet: import_zod29.z.boolean(),
  useTiers: import_zod29.z.boolean(),
  viewMode: import_zod29.z.string()
});
var view_settings_schema_default = conditionalStrict(ViewSettingsSchema);

// src/schema/deck-list/deck-list.schema.ts
var DeckListSchema = import_zod30.z.object({
  affiliates: affiliates_schema_default,
  allowPrimerClone: import_zod30.z.boolean(),
  areCommentsEnabled: import_zod30.z.boolean(),
  authors: import_zod30.z.array(author_schema_default),
  authorsCanEdit: import_zod30.z.boolean(),
  authorTags: author_tags_schema_default,
  autoBracket: import_zod30.z.number().nullish(),
  boards: boards_schema_default,
  bookmarkCount: import_zod30.z.coerce.number().nullish(),
  bracket: import_zod30.z.number().nullish(),
  cardsToTokens: cards_to_tokens_schema_default,
  colorIdentity: import_zod30.z.array(color_identity_schema_default),
  colorIdentityPercentages: color_identity_percentages_schema_default,
  colorPercentages: color_percentages_schema_default,
  colors: import_zod30.z.array(color_identity_schema_default),
  commentCount: import_zod30.z.number(),
  createdAtUtc: import_zod30.z.coerce.date(),
  createdByUser: author_schema_default,
  description: import_zod30.z.string(),
  enableMultiplePrintings: import_zod30.z.boolean(),
  exportId: import_zod30.z.uuid(),
  folder: folder_schema_default.nullish(),
  format: import_zod30.z.string(),
  hubs: import_zod30.z.array(hub_schema_default),
  id: import_zod30.z.string(),
  ignoreBrackets: import_zod30.z.boolean(),
  includeBasicLandsInPrice: import_zod30.z.boolean(),
  includeCommandersInPrice: import_zod30.z.boolean(),
  includeSignatureSpellsInPrice: import_zod30.z.boolean(),
  isMature: import_zod30.z.boolean().nullish(),
  isMatureAuto: import_zod30.z.boolean().nullish(),
  isMatureMod: import_zod30.z.boolean().nullish(),
  isShared: import_zod30.z.boolean(),
  isTooBeaucoup: import_zod30.z.boolean(),
  lastUpdatedAtUtc: import_zod30.z.coerce.date(),
  likeCount: import_zod30.z.number(),
  main: card_schema_default.nullish(),
  mainCardIdIsBackFace: import_zod30.z.boolean(),
  media: import_zod30.z.array(import_zod30.z.never()),
  name: import_zod30.z.string(),
  originalDeck: original_deck_schema_default.nullish(),
  ownerUserId: import_zod30.z.string(),
  publicId: import_zod30.z.string(),
  publicUrl: import_zod30.z.url(),
  requestedAuthors: import_zod30.z.array(author_schema_default),
  sfwCommentCount: import_zod30.z.number(),
  tokenMappings: token_mappings_schema_default,
  tokens: import_zod30.z.array(card_schema_default),
  tokensToCards: tokens_to_cards_schema_default,
  userBracket: import_zod30.z.number().nullish(),
  version: import_zod30.z.number(),
  viewCount: import_zod30.z.number(),
  viewSettings: view_settings_schema_default.nullish(),
  visibility: visibility_schema_default
});
var deck_list_schema_default = conditionalStrict(DeckListSchema);

// src/api/deck-list/deck-list.api.ts
var MOXFIELD_DECK_URL_PREFIX = "https://moxfield.com/decks/";
var MOXFIELD_DECK_URL_PREFIX_WWW = "https://www.moxfield.com/decks/";
var API_BASE_URL = "https://api2.moxfield.com";
var deckListIdSchema = import_zod31.z.coerce.string().describe("DeckList id").transform((value) => {
  const parsed = import_zod31.z.url().safeParse(value);
  if (parsed.error) {
    return value;
  }
  const url = parsed.data;
  if (url.startsWith(MOXFIELD_DECK_URL_PREFIX) || url.startsWith(MOXFIELD_DECK_URL_PREFIX_WWW)) {
    const replaced = url.replace(MOXFIELD_DECK_URL_PREFIX, "").replace(MOXFIELD_DECK_URL_PREFIX_WWW, "");
    const split = replaced.split("/");
    return split.at(0);
  }
  return parsed.data;
});
var fetchWithZod = (0, import_zod_fetch.createZodFetcher)(createMoxfieldFetcher());
var DeckListApi = class {
  async findById(id) {
    const parsed = deckListIdSchema.parse(id);
    return fetchWithZod(deck_list_schema_default, `${API_BASE_URL}/v3/decks/all/${parsed}`, {
      method: "GET"
    });
  }
};
var deck_list_api_default = DeckListApi;

// src/api/index.ts
var MoxfieldApi = class {
  deckList;
  constructor() {
    this.deckList = new deck_list_api_default();
  }
};
var api_default = MoxfieldApi;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AffiliatesSchema,
  AuthorSchema,
  AuthorTagsSchema,
  BoardCardSchema,
  BoardSchema,
  BoardTypeSchema,
  BoardsSchema,
  CardFaceSchema,
  CardSchema,
  CardsToTokensSchema,
  ColorIdentityPercentagesSchema,
  ColorIdentitySchema,
  ColorPercentagesSchema,
  DeckListSchema,
  FinishSchema,
  FolderSchema,
  HubSchema,
  LegalitiesSchema,
  MeldPartSchema,
  MeldResultSchema,
  MoxfieldError,
  NotFoundMoxfieldError,
  OriginalDeckSchema,
  PricesSchema,
  PrintCardSchema,
  RaritySchema,
  SetTypeSchema,
  TokenMappingsSchema,
  TokensToCardsSchema,
  ViewSettingsSchema,
  VisibilitySchema
});
