// moxfield-api/dist/moxfield-api.js
// src/api/deck-list/deck-list.api.ts
import { z as z31 } from "zod";
import { createZodFetcher } from "zod-fetch";

// src/error/moxfield.error.ts
var MoxfieldError = class extends Error {
  constructor(status, message) {
    super(message);
    this.status = status;
  }
};

// src/error/not-found.error.ts
var NotFoundMoxfieldError = class extends MoxfieldError {
  constructor(message) {
    super(404, message);
  }
};

// src/fetcher/moxfield.fetcher.ts
function createMoxfieldFetcher(fetcher = fetch) {
  return async function moxfieldFetcher(...arguments_) {
    const response = await fetcher(...arguments_);
    if (response.ok) {
      return response.json();
    }
    if (response.status === 404) {
      throw new NotFoundMoxfieldError();
    }
    const object = await response.json();
    const status = "status" in object ? Number(object.status) : void 0;
    const error = "error" in object ? String(object.error) : void 0;
    throw new MoxfieldError(status, error);
  };
}

// src/schema/affiliates/affiliates.schema.ts
import { z as z2 } from "zod";

// src/helpers/is-test.ts
function isTest() {
  return typeof process !== "undefined" && process.env.NODE_ENV === "test";
}

// src/helpers/strict.ts
import z from "zod";
function conditionalStrict(schema) {
  if (isTest()) {
    return z.strictObject(schema.shape);
  }
  return schema;
}

// src/schema/affiliates/affiliates.schema.ts
var AffiliatesSchema = z2.object({
  ch: z2.string(),
  ck: z2.string(),
  cm: z2.string(),
  csi: z2.string(),
  ct: z2.string(),
  scg: z2.string(),
  tcg: z2.string()
});
var affiliates_schema_default = conditionalStrict(AffiliatesSchema);

// src/schema/author/author.schema.ts
import { z as z3 } from "zod";
var AuthorSchema = z3.object({
  badges: z3.array(z3.string()),
  displayName: z3.string(),
  profileImageUrl: z3.url().nullish(),
  promos: z3.array(z3.unknown()).nullish(),
  userName: z3.string()
});
var author_schema_default = conditionalStrict(AuthorSchema);

// src/schema/author-tags/author-tags.schema.ts
import { z as z4 } from "zod";
var AuthorTagsSchema = z4.record(z4.string(), z4.array(z4.string()));
var author_tags_schema_default = AuthorTagsSchema;

// src/schema/board/board.schema.ts
import { z as z18 } from "zod";

// src/schema/board-card/board-card.schema.ts
import { z as z17 } from "zod";

// src/schema/board-type/board-type.schema.ts
import { z as z5 } from "zod";
var BoardTypesSchema = z5.union([
  z5.literal("attractions"),
  z5.literal("commanders"),
  z5.literal("companions"),
  z5.literal("contraptions"),
  z5.literal("mainboard"),
  z5.literal("maybeboard"),
  z5.literal("partners"),
  z5.literal("planes"),
  z5.literal("schemes"),
  z5.literal("sideboard"),
  z5.literal("signatureSpells"),
  z5.literal("stickers"),
  z5.literal("tokens")
]);
var board_type_schema_default = BoardTypesSchema;

// src/schema/card/card.schema.ts
import { z as z14 } from "zod";

// src/schema/card-face/card-face.schema.ts
import { z as z7 } from "zod";

// src/schema/color-identity/color-identity.schema.ts
import { z as z6 } from "zod";
var ColorIdentitySchema = z6.union([z6.literal("W"), z6.literal("U"), z6.literal("B"), z6.literal("R"), z6.literal("G")]);
var color_identity_schema_default = ColorIdentitySchema;

// src/schema/card-face/card-face.schema.ts
var CardFaceSchema = z7.object({
  color_indicator: z7.array(color_identity_schema_default),
  colors: z7.array(color_identity_schema_default),
  defense: z7.string().nullish(),
  flavor_name: z7.string().nullish(),
  flavor_text: z7.string().nullish(),
  id: z7.string(),
  image_seq: z7.number(),
  loyalty: z7.string().nullish(),
  mana_cost: z7.string().nullish(),
  name: z7.string(),
  oracle_text: z7.string().nullish(),
  power: z7.string().nullish(),
  toughness: z7.string().nullish(),
  type_line: z7.string()
});
var card_face_schema_default = conditionalStrict(CardFaceSchema);

// src/schema/legalities/legalities.schema.ts
import { z as z8 } from "zod";
var LegalitiesFormatSchema = z8.union([
  z8.literal("alchemy"),
  z8.literal("brawl"),
  z8.literal("commander"),
  z8.literal("duel"),
  z8.literal("explorer"),
  z8.literal("future"),
  z8.literal("gladiator"),
  z8.literal("historic"),
  z8.literal("historicbrawl"),
  z8.literal("legacy"),
  z8.literal("modern"),
  z8.literal("oathbreaker"),
  z8.literal("oldschool"),
  z8.literal("pauper"),
  z8.literal("paupercommander"),
  z8.literal("penny"),
  z8.literal("pioneer"),
  z8.literal("predh"),
  z8.literal("premodern"),
  z8.literal("standard"),
  z8.literal("standardbrawl"),
  z8.literal("timeless"),
  z8.literal("vintage")
]);
var LegalitiesValueSchema = z8.union([
  z8.literal("legal"),
  z8.literal("not_legal"),
  z8.literal("banned"),
  z8.literal("restricted")
]);
var LegalitiesSchema = z8.partialRecord(LegalitiesFormatSchema, LegalitiesValueSchema);
var legalities_schema_default = LegalitiesSchema;

// src/schema/meld-part/meld-part.schema.ts
import { z as z9 } from "zod";
var MeldPartSchema = z9.object({
  id: z9.string(),
  imageCardId: z9.string(),
  imageCardIdIsCardFace: z9.boolean(),
  name: z9.string(),
  uniqueCardId: z9.string()
});
var meld_part_schema_default = conditionalStrict(MeldPartSchema);

// src/schema/meld-result/meld-result.schema.ts
import { z as z10 } from "zod";
var MeldResultSchema = z10.object({
  id: z10.string(),
  imageCardId: z10.string(),
  imageCardIdIsCardFace: z10.boolean(),
  name: z10.string(),
  uniqueCardId: z10.string()
});
var meld_result_schema_default = conditionalStrict(MeldResultSchema);

// src/schema/prices/prices.schema.ts
import { z as z11 } from "zod";
var PricesSchema = z11.object({
  ck: z11.number().nullish(),
  ck_buy: z11.number().nullish(),
  ck_buy_etched: z11.number().nullish(),
  ck_buy_etched_qty: z11.number().nullish(),
  ck_buy_foil: z11.number().nullish(),
  ck_buy_foil_qty: z11.number().nullish(),
  ck_buy_qty: z11.number().nullish(),
  ck_etched: z11.number().nullish(),
  ck_foil: z11.number().nullish(),
  csi: z11.number().nullish(),
  csi_buy: z11.number().nullish(),
  csi_buy_foil: z11.number().nullish(),
  csi_buy_foil_qty: z11.number().nullish(),
  csi_buy_qty: z11.number().nullish(),
  csi_foil: z11.number().nullish(),
  ct: z11.number().nullish(),
  ct_etched: z11.number().nullish(),
  ct_foil: z11.number().nullish(),
  eur: z11.number().nullish(),
  eur_foil: z11.number().nullish(),
  lastUpdatedAtUtc: z11.coerce.date(),
  mp: z11.number().nullish(),
  mp_etched: z11.number().nullish(),
  mp_foil: z11.number().nullish(),
  mp_qty: z11.number().nullish(),
  scg: z11.number().nullish(),
  scg_buy: z11.number().nullish(),
  scg_etched: z11.number().nullish(),
  scg_etched_buy: z11.number().nullish(),
  scg_etched_qty: z11.number().nullish(),
  scg_foil: z11.number().nullish(),
  scg_foil_buy: z11.number().nullish(),
  scg_foil_qty: z11.number().nullish(),
  scg_qty: z11.number().nullish(),
  tix: z11.number().nullish(),
  usd: z11.number().nullish(),
  usd_etched: z11.number().nullish(),
  usd_foil: z11.number().nullish()
});
var prices_schema_default = conditionalStrict(PricesSchema);

// src/schema/rarity/rarity.schema.ts
import { z as z12 } from "zod";
var RaritySchema = z12.union([
  z12.literal("bonus"),
  z12.literal("common"),
  z12.literal("mythic"),
  z12.literal("rare"),
  z12.literal("special"),
  z12.literal("uncommon")
]);
var rarity_schema_default = RaritySchema;

// src/schema/set-type/set-type.schema.ts
import { z as z13 } from "zod";
var SetTypeSchema = z13.union([
  z13.literal("alchemy"),
  z13.literal("archenemy"),
  z13.literal("arsenal"),
  z13.literal("box"),
  z13.literal("commander"),
  z13.literal("core"),
  z13.literal("draft_innovation"),
  z13.literal("duel_deck"),
  z13.literal("eternal"),
  z13.literal("expansion"),
  z13.literal("from_the_vault"),
  z13.literal("funny"),
  z13.literal("masterpiece"),
  z13.literal("masters"),
  z13.literal("memorabilia"),
  z13.literal("minigame"),
  z13.literal("planechase"),
  z13.literal("premium_deck"),
  z13.literal("promo"),
  z13.literal("spellbook"),
  z13.literal("starter"),
  z13.literal("token"),
  z13.literal("treasure_chest"),
  z13.literal("vanguard")
]);
var set_type_schema_default = SetTypeSchema;

// src/schema/card/card.schema.ts
var CardSchema = z14.object({
  acorn: z14.boolean(),
  allRaritiesMask: z14.number().nullish(),
  arena_id: z14.number().nullish(),
  artist: z14.string().nullish(),
  attraction_lights: z14.array(z14.number()).nullish(),
  border_color: z14.string(),
  card_faces: z14.array(card_face_schema_default),
  cardHoarderUrl: z14.url().nullish(),
  cardkingdom_etched_id: z14.number().nullish(),
  cardkingdom_foil_id: z14.number().nullish(),
  cardkingdom_id: z14.number().nullish(),
  cardKingdomEtchedUrl: z14.string().nullish(),
  cardKingdomFoilUrl: z14.string().nullish(),
  cardKingdomUrl: z14.string().nullish(),
  cardmarket_id: z14.number().nullish(),
  cardMarketUrl: z14.url().nullish(),
  cardTraderEtchedUrl: z14.url().nullish(),
  cardTraderFoilUrl: z14.url().nullish(),
  cardTraderUrl: z14.url().nullish(),
  cmc: z14.number(),
  cn: z14.string(),
  color_identity: z14.array(color_identity_schema_default),
  color_indicator: z14.array(color_identity_schema_default),
  colors: z14.array(color_identity_schema_default),
  colorshifted: z14.boolean(),
  content_warning: z14.boolean(),
  coolStuffIncFoilUrl: z14.url().nullish(),
  coolStuffIncUrl: z14.url().nullish(),
  defaultFinish: z14.string(),
  digital: z14.boolean(),
  edhrec_rank: z14.number().nullish(),
  etched: z14.boolean(),
  flavor_name: z14.string().nullish(),
  flavor_text: z14.string().nullish(),
  foil: z14.boolean(),
  frame: z14.string().nullish(),
  frame_effects: z14.array(z14.string()).nullish(),
  glossy: z14.boolean(),
  has_arena_legal: z14.boolean(),
  has_multiple_editions: z14.boolean(),
  id: z14.string(),
  image_seq: z14.number(),
  isArenaLegal: z14.boolean(),
  isCovered: z14.boolean(),
  isPauperCommander: z14.boolean(),
  isToken: z14.boolean(),
  lang: z14.string(),
  latest: z14.boolean(),
  layout: z14.string(),
  legalities: legalities_schema_default,
  loyalty: z14.string().nullish(),
  mana_cost: z14.string().nullish(),
  manapool_url: z14.url().nullish(),
  meld_part: meld_part_schema_default.nullish(),
  meld_parts: z14.array(meld_part_schema_default).nullish(),
  meld_result: meld_result_schema_default.nullish(),
  mtgo_id: z14.number().nullish(),
  multiverse_ids: z14.array(z14.number()).nullish(),
  name: z14.string(),
  nonfoil: z14.boolean(),
  oracle_text: z14.string().nullish(),
  power: z14.string().nullish(),
  prices: prices_schema_default,
  printed_name: z14.string().nullish(),
  printed_text: z14.string().nullish(),
  printed_type_line: z14.string().nullish(),
  promo_types: z14.array(z14.string()),
  rarity: rarity_schema_default,
  released_at: z14.coerce.date().nullish(),
  reprint: z14.boolean(),
  reserved: z14.boolean(),
  scryfall_id: z14.uuid(),
  set: z14.string(),
  set_name: z14.string(),
  set_type: set_type_schema_default.nullish(),
  starcitygames_etched_sku: z14.string().nullish(),
  starcitygames_etched_url: z14.url().nullish(),
  starcitygames_foil_sku: z14.string().nullish(),
  starcitygames_foil_url: z14.url().nullish(),
  starcitygames_sku: z14.string().nullish(),
  starcitygames_url: z14.url().nullish(),
  tcgplayer_id: z14.number().nullish(),
  tcgPlayerUrl: z14.url().nullish(),
  toughness: z14.string().nullish(),
  type: z14.string().nullish(),
  type_line: z14.string().nullish(),
  uniqueCardId: z14.string()
});
var card_schema_default = conditionalStrict(CardSchema);

// src/schema/finish/finish.schema.ts
import { z as z15 } from "zod";
var FinishSchema = z15.union([z15.literal("nonFoil"), z15.literal("foil"), z15.literal("etched"), z15.literal("glossy")]);
var finish_schema_default = FinishSchema;

// src/schema/print-card/print-card.schema.ts
import { z as z16 } from "zod";
var PrintCardSchema = z16.object({
  card: card_schema_default,
  cardHoarderUrl: z16.url().nullish(),
  cardKingdomFoilUrl: z16.string().nullish(),
  cardKingdomUrl: z16.string().nullish(),
  cardMarketUrl: z16.url().nullish(),
  cn: z16.string(),
  coolStuffIncFoilUrl: z16.url().nullish(),
  coolStuffIncUrl: z16.url().nullish(),
  finish: finish_schema_default,
  foil: z16.boolean(),
  hash: z16.string(),
  id: z16.string(),
  imageCardId: z16.string(),
  imageCardIdIsCardFace: z16.boolean(),
  isAlter: z16.boolean(),
  isFoil: z16.boolean(),
  isProxy: z16.boolean(),
  lang: z16.string(),
  nonfoil: z16.boolean(),
  prices: prices_schema_default,
  quantity: z16.number(),
  rarity: rarity_schema_default,
  set: z16.string(),
  set_name: z16.string(),
  tcgPlayerUrl: z16.url().nullish()
});
var print_card_schema_default = conditionalStrict(PrintCardSchema);

// src/schema/board-card/board-card.schema.ts
var BoardCardSchema = z17.object({
  boardType: board_type_schema_default,
  card: card_schema_default,
  cmcOverride: z17.number().nullish(),
  colorIdentityOverride: z17.array(color_identity_schema_default).nullish(),
  excludedFromColor: z17.boolean(),
  finish: finish_schema_default,
  isAlter: z17.boolean(),
  isCompanion: z17.boolean().nullish(),
  isFoil: z17.boolean(),
  isProxy: z17.boolean(),
  manaCostOverride: z17.string().nullish(),
  printingData: z17.array(print_card_schema_default).nullish(),
  quantity: z17.number(),
  useCmcOverride: z17.boolean(),
  useColorIdentityOverride: z17.boolean(),
  useManaCostOverride: z17.boolean()
});
var board_card_schema_default = conditionalStrict(BoardCardSchema);

// src/schema/board/board.schema.ts
var BoardSchema = z18.object({
  cards: z18.record(z18.string(), board_card_schema_default),
  count: z18.number()
});
var board_schema_default = conditionalStrict(BoardSchema);

// src/schema/boards/boards.schema.ts
import { z as z19 } from "zod";
var BoardTypesSchema2 = z19.union([
  z19.literal("attractions"),
  z19.literal("commanders"),
  z19.literal("companions"),
  z19.literal("contraptions"),
  z19.literal("mainboard"),
  z19.literal("maybeboard"),
  z19.literal("planes"),
  z19.literal("schemes"),
  z19.literal("sideboard"),
  z19.literal("signatureSpells"),
  z19.literal("stickers"),
  z19.literal("tokens")
]);
var BoardsSchema = z19.record(BoardTypesSchema2, board_schema_default);
var boards_schema_default = BoardsSchema;

// src/schema/cards-to-tokens/cards-to-tokens.schema.ts
import { z as z20 } from "zod";
var CardsToTokensSchema = z20.record(z20.string(), z20.array(z20.string()));
var cards_to_tokens_schema_default = CardsToTokensSchema;

// src/schema/color-identity-percentages/color-identity-percentages.schema.ts
import { z as z21 } from "zod";
var ColorIdentityPercentagesSchema = z21.object({
  black: z21.number(),
  blue: z21.number(),
  green: z21.number(),
  red: z21.number(),
  white: z21.number()
});
var color_identity_percentages_schema_default = conditionalStrict(ColorIdentityPercentagesSchema);

// src/schema/color-percentages/color-percentages.schema.ts
import { z as z22 } from "zod";
var ColorPercentagesSchema = z22.object({
  black: z22.number(),
  blue: z22.number(),
  green: z22.number(),
  red: z22.number(),
  white: z22.number()
});
var color_percentages_schema_default = conditionalStrict(ColorPercentagesSchema);

// src/schema/deck-list/deck-list.schema.ts
import { z as z30 } from "zod";

// src/schema/folder/folder.schema.ts
import { z as z23 } from "zod";
var FolderSchema = z23.object({
  id: z23.string(),
  name: z23.string()
});
var folder_schema_default = conditionalStrict(FolderSchema);

// src/schema/hub/hub.schema.ts
import { z as z24 } from "zod";
var HubSchema = z24.object({
  description: z24.string(),
  name: z24.string()
});
var hub_schema_default = conditionalStrict(HubSchema);

// src/schema/original-deck/original-deck.schema.ts
import { z as z26 } from "zod";

// src/schema/visibility/visibility.schema.ts
import { z as z25 } from "zod";
var VisibilitySchema = z25.union([
  z25.literal("unlisted"),
  z25.literal("private"),
  z25.literal("public"),
  z25.literal("deleted")
]);
var visibility_schema_default = VisibilitySchema;

// src/schema/original-deck/original-deck.schema.ts
var OriginalDeckSchema = z26.object({
  authors: z26.array(author_schema_default),
  createdAtUtc: z26.coerce.date(),
  createdByUser: author_schema_default,
  id: z26.string(),
  lastUpdatedAtUtc: z26.coerce.date(),
  mainCardId: z26.string().nullish(),
  mainCardIdIsBackFace: z26.boolean(),
  mainCardIdIsCardFace: z26.boolean(),
  name: z26.string(),
  publicId: z26.string(),
  visibility: visibility_schema_default
});
var original_deck_schema_default = conditionalStrict(OriginalDeckSchema);

// src/schema/token-mappings/token-mappings.schema.ts
import { z as z27 } from "zod";
var TokenMappingsSchema = z27.record(z27.string(), card_schema_default);
var token_mappings_schema_default = TokenMappingsSchema;

// src/schema/tokens-to-cards/tokens-to-cards.schema.ts
import { z as z28 } from "zod";
var TokensToCardsSchema = z28.record(z28.string(), z28.array(z28.string()));
var tokens_to_cards_schema_default = TokensToCardsSchema;

// src/schema/view-settings/view-settings.schema.ts
import { z as z29 } from "zod";
var ViewSettingsSchema = z29.object({
  allowMultiplePrintings: z29.boolean(),
  groupBy: z29.string(),
  sortBy: z29.string(),
  useMana: z29.boolean(),
  usePrice: z29.boolean(),
  useSet: z29.boolean(),
  useTiers: z29.boolean(),
  viewMode: z29.string()
});
var view_settings_schema_default = conditionalStrict(ViewSettingsSchema);

// src/schema/deck-list/deck-list.schema.ts
var DeckListSchema = z30.object({
  affiliates: affiliates_schema_default,
  allowPrimerClone: z30.boolean(),
  areCommentsEnabled: z30.boolean(),
  authors: z30.array(author_schema_default),
  authorsCanEdit: z30.boolean(),
  authorTags: author_tags_schema_default,
  autoBracket: z30.number().nullish(),
  boards: boards_schema_default,
  bookmarkCount: z30.coerce.number().nullish(),
  bracket: z30.number().nullish(),
  cardsToTokens: cards_to_tokens_schema_default,
  colorIdentity: z30.array(color_identity_schema_default),
  colorIdentityPercentages: color_identity_percentages_schema_default,
  colorPercentages: color_percentages_schema_default,
  colors: z30.array(color_identity_schema_default),
  commentCount: z30.number(),
  createdAtUtc: z30.coerce.date(),
  createdByUser: author_schema_default,
  description: z30.string(),
  enableMultiplePrintings: z30.boolean(),
  exportId: z30.uuid(),
  folder: folder_schema_default.nullish(),
  format: z30.string(),
  hubs: z30.array(hub_schema_default),
  id: z30.string(),
  ignoreBrackets: z30.boolean(),
  includeBasicLandsInPrice: z30.boolean(),
  includeCommandersInPrice: z30.boolean(),
  includeSignatureSpellsInPrice: z30.boolean(),
  isMature: z30.boolean().nullish(),
  isMatureAuto: z30.boolean().nullish(),
  isMatureMod: z30.boolean().nullish(),
  isShared: z30.boolean(),
  isTooBeaucoup: z30.boolean(),
  lastUpdatedAtUtc: z30.coerce.date(),
  likeCount: z30.number(),
  main: card_schema_default.nullish(),
  mainCardIdIsBackFace: z30.boolean(),
  media: z30.array(z30.never()),
  name: z30.string(),
  originalDeck: original_deck_schema_default.nullish(),
  ownerUserId: z30.string(),
  publicId: z30.string(),
  publicUrl: z30.url(),
  requestedAuthors: z30.array(author_schema_default),
  sfwCommentCount: z30.number(),
  tokenMappings: token_mappings_schema_default,
  tokens: z30.array(card_schema_default),
  tokensToCards: tokens_to_cards_schema_default,
  userBracket: z30.number().nullish(),
  version: z30.number(),
  viewCount: z30.number(),
  viewSettings: view_settings_schema_default.nullish(),
  visibility: visibility_schema_default
});
var deck_list_schema_default = conditionalStrict(DeckListSchema);

// src/api/deck-list/deck-list.api.ts
var MOXFIELD_DECK_URL_PREFIX = "https://moxfield.com/decks/";
var MOXFIELD_DECK_URL_PREFIX_WWW = "https://www.moxfield.com/decks/";
var API_BASE_URL = "https://api2.moxfield.com";
var deckListIdSchema = z31.coerce.string().describe("DeckList id").transform((value) => {
  const parsed = z31.url().safeParse(value);
  if (parsed.error) {
    return value;
  }
  const url = parsed.data;
  if (url.startsWith(MOXFIELD_DECK_URL_PREFIX) || url.startsWith(MOXFIELD_DECK_URL_PREFIX_WWW)) {
    const replaced = url.replace(MOXFIELD_DECK_URL_PREFIX, "").replace(MOXFIELD_DECK_URL_PREFIX_WWW, "");
    const split = replaced.split("/");
    return split.at(0);
  }
  return parsed.data;
});
var fetchWithZod = createZodFetcher(createMoxfieldFetcher());
var DeckListApi = class {
  async findById(id) {
    const parsed = deckListIdSchema.parse(id);
    return fetchWithZod(deck_list_schema_default, `${API_BASE_URL}/v3/decks/all/${parsed}`, {
      method: "GET"
    });
  }
};
var deck_list_api_default = DeckListApi;

// src/api/index.ts
var MoxfieldApi = class {
  deckList;
  constructor() {
    this.deckList = new deck_list_api_default();
  }
};
var api_default = MoxfieldApi;
export {
  affiliates_schema_default as AffiliatesSchema,
  author_schema_default as AuthorSchema,
  author_tags_schema_default as AuthorTagsSchema,
  board_card_schema_default as BoardCardSchema,
  board_schema_default as BoardSchema,
  board_type_schema_default as BoardTypeSchema,
  boards_schema_default as BoardsSchema,
  card_face_schema_default as CardFaceSchema,
  card_schema_default as CardSchema,
  cards_to_tokens_schema_default as CardsToTokensSchema,
  color_identity_percentages_schema_default as ColorIdentityPercentagesSchema,
  color_identity_schema_default as ColorIdentitySchema,
  color_percentages_schema_default as ColorPercentagesSchema,
  deck_list_schema_default as DeckListSchema,
  finish_schema_default as FinishSchema,
  folder_schema_default as FolderSchema,
  hub_schema_default as HubSchema,
  legalities_schema_default as LegalitiesSchema,
  meld_part_schema_default as MeldPartSchema,
  meld_result_schema_default as MeldResultSchema,
  MoxfieldError,
  NotFoundMoxfieldError,
  original_deck_schema_default as OriginalDeckSchema,
  prices_schema_default as PricesSchema,
  print_card_schema_default as PrintCardSchema,
  rarity_schema_default as RaritySchema,
  set_type_schema_default as SetTypeSchema,
  token_mappings_schema_default as TokenMappingsSchema,
  tokens_to_cards_schema_default as TokensToCardsSchema,
  view_settings_schema_default as ViewSettingsSchema,
  visibility_schema_default as VisibilitySchema,
  api_default as default
};
